### ISSUE TITLE
dSYM upload not working

### ISSUE THREAD

**Issue by djbe** [getsentry/onpremise#328]
Hi,

I seem to be having a similar issue to #245. Running `sentry-cli` (1.49.0) either gives:

> \> Found 31 debug information files
> \> Prepared debug information files for upload
> \> Nothing to upload, all files are on the server

Or:

> \> Found 31 debug information files
> \> Prepared debug information files for upload
> \> File upload complete:
>
>  PENDING 1a729d81-e618-3d5e-9e53-c9575ea53656 (1a729d81-e618-3d5e-9e53-c9575ea53656.dSYM/Contents/Resources/DWARF/FetchedDataSource; arm64 debug companion)
>  PENDING 32b22a78-4559-319a-b6af-545cc9994c7d (32b22a78-4559-319a-b6af-545cc9994c7d.dSYM/Contents/Resources/DWARF/p2_OAuth2; arm64 debug companion)
>  ... and a bunch more, 31 in total ...

Checking on server, they are not added to the list of uploaded debug information files, and "Pending Issues" keeps complaining about missing debug files. I've also noticed the `internal` logger has been logging about some issue, ever since this started:

https://sentry.appwi.se/share/issue/c9119d901bac40f29dd3cc534885d5be/
If you need more info than the linked issue, please let me know!

As mentioned in #245, I tried running `... web upgrade` and `... web repair`, to no avail. This is all on Sentry on premise, having run the `install.sh` script again today (10 minutes ago) just to be sure. The installed version before that was also 10.0.0dev dating from a few days ago.

**Comment by BYK** [comment]
Hi, thanks a lot for the report! I'm pretty sure this is related to #295. There is some updated config in the `config.yml` file: https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml#L56-L60

If you ever installed Sentry from on-premise beta before, that means you have a stale `config.yml` file and didn't get these updates. The install script doesn't touch or tries to do any kind of reconciliation if there's already a config file, assuming the user changed them. We can probably handle this better with file signatures etc but we expect people to check the configs during installs and upgrades.

Long story short, can you check your `config.yml` file for the file settings, make sure they are aligned with the one from the repo and then run `./install.sh` again? (alternatively you can do

```bash
docker-compose down
docker-compose build web
docker-compose build sentry-cleanup
docker-compose up -d
```

**Comment by djbe** [comment]
Jep, this is what I had:

**Comment by djbe** [comment]
Allright, when I try to upload a dSym, now I get this:

> \> Found 31 debug information files
> \> Prepared debug information files for upload
> error: API request failed
>   caused by: sentry reported an error: request failure (http status: 500)
>   Object({"detail": String("Internal Error"), "errorId": String("94e5bf544e51498e9918371572c823ee")})

The corresponding error in "internal":
https://sentry.appwi.se/share/issue/eba56c95c5c348998badc13e841c5c77/

Besides the config change, do I need to change anything else in on the server?

**Comment by BYK** [comment]
Ugh, sorry for all the trouble. For some reason this didn't seem to work for the ownership changes: https://github.com/getsentry/onpremise/blob/b2080c14467fd44d10c3faffc7efc80a9d98c96d/install.sh#L150-L156

Can you try running the following:

```bash
docker-compose run --rm --entrypoint /bin/bash web -c "chown -R sentry:sentry /data"
```

It may take a while if you have too many files there but it will finish and should resolve the issue.

If it doesn't I'd run the following afterwards:

```bash
docker-compose run --rm --entrypoint /bin/bash web -c "chmod -R 0770 /data"
```

Great use of the Sentry issue sharing btw, very helpful!

**Comment by djbe** [comment]
Aha, that first one (chown) fixed it! 

Big thanks for the quick & helpful responses! I'm still somewhat unfamiliar with docker, and still need to wrap my head around executing commands in there. I kept looking for some folder in the normal filesystem 😅

**Comment by BYK** [comment]
@djbe glad it is resolved! I wonder if we should figure out a way to warn people when there are updates to these default configurations. Also we had protections in place for this but they had incorrect assumptions so I submitted a fix for that: getsentry/sentry#16353.

Thanks a lot for all the help!

**Comment by djbe** [comment]
Well, it was a beta so I knew stuff could go wrong 😉

Maybe storing a copy of the example config files somewhere on install, and diffing that when running install itself, throwing a big warning to let users know things changed?

**Comment by BYK** [comment]
I think we can store some signatures in the file and use them to verify if the file was touched by the user or not:

1. If the user haven't touched the file at all, simply override it with the example, hassle-free.
2. If the user touched the file but the version in the repo hasn't changed, stay silent.
3. If the user touched the file _and_ the version in the repo changed, halt, show a warning with a diff prompt or something forcing the user to take an action. Maybe try an automatic merge first.

What do you think?

**Comment by djbe** [comment]
Sounds good, although I'm not sure where you're going to store the signatures? In the repo and gitignore it?

**Comment by farfromrefug** [comment]
@BYK how can i use your latest changes. From the docker registry i dont see 10.0.0dev
